agent/mapcall/src/main/java/randoop/instrument/MethodSignature.java

```diff
@@ -0,0 +1,275 @@
+    if (!paramString.isEmpty()) {
+      parameters = paramString.split("\\s*,\\s*");
+    }
+    return MethodSignature.of(name, parameters);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof MethodSignature)) {
+      return false;
+    }
+    MethodSignature md = (MethodSignature) obj;
+    return this.classname.equals(md.classname)
+        && this.name.equals(md.name)
+        && Arrays.equals(this.paramTypes, md.paramTypes);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(classname, name, Arrays.hashCode(paramTypes));
+  }
+
+  /**
+   * Returns the fully-qualified signature string for this {@link MethodSignature}.
+   *
+   * @return the fully-qualified signature string for this {@link MethodSignature}
+   */
+  @Override
+  public String toString() {
+    return String.format("%s.%s(%s)", classname, name, UtilMDE.join(paramTypes, ", "));
+  }
+
+  /**
+   * Returns the fully-qualified class name of this {@link MethodSignature}.
+   *
+   * @return the fully-qualified class name of this {@link MethodSignature}
+   */
+  String getClassname() {
+    return classname;
+  }
+
+  /**
+   * Returns the method name for this {@link MethodSignature}.
+   *
+   * @return the method name of this {@link MethodSignature}
+   */
+  String getName() {
+    return name;
+  }
+
+  /**
+   * Returns the parameter types (as BCEL {@code Type} references) for this {@link MethodSignature}.
+   *
+   * @return the parameter types for this {@link MethodSignature}
+   */
+  Type[] getParameterTypes() {
+    return paramTypes;
+  }
+
+  /**
+   * Uses reflection to get the {@code java.lang.reflect.Method} object for this {@link
+   * MethodSignature}.
+   *
+   * @return the reflection object for this {@link MethodSignature}
+   * @throws ClassNotFoundException if the containing class of this {@link MethodSignature} is not
+   *     found on the classpath
+   * @throws NoSuchMethodException if the containing class of this {@link MethodSignature} does not
+   *     have the represented method as a member
+   */
+  Method toMethod() throws ClassNotFoundException, NoSuchMethodException {
+    if (method != null) {
+      return method;
+    }
+
+    Class<?> methodClass = Class.forName(classname);
+    Class<?> params[] = new Class[paramTypes.length];
+    for (int i = 0; i < paramTypes.length; i++) {
+      try {
+        params[i] = typeToClass(paramTypes[i]);
+      } catch (ClassNotFoundException e) {
+        throw new ArgumentClassNotFoundException(e.getMessage());
+      }
+    }
+
+    // First check it the method is declared in the class
+    try {
+      method = methodClass.getDeclaredMethod(name, params);
+      method.setAccessible(true);
+      return method;
+    } catch (NoSuchMethodException e) {
+      // ignore -- look for inherited method
+    }
+
+    // If it is not declared in class, check if it is inherited
+    method = methodClass.getMethod(name, params);
```

Comment:
`getMethod` only returns the public methods of the class. 

We ignore private methods but I don't think we want to make that restriction here.